src/hdmf/common/resources.py

```diff
@@ -339,47 +363,54 @@ def _check_object_field(self, container, relative_path, field, create):
              'doc': ('The field of the compound data type using an external resource.')})
     def get_key(self, **kwargs):
         """
-        Return a Key or a list of Key objects that correspond to the given key.
+        Return a Key.
 
         If container, relative_path, and field are provided, the Key that corresponds to the given name of the key
         for the given container, relative_path, and field is returned.
         """
         key_name, container, relative_path, field = popargs('key_name', 'container', 'relative_path', 'field', kwargs)
         key_idx_matches = self.keys.which(key=key_name)
 
+        file = kwargs['file']
+
         if container is not None:
+            if file is None:
+                if isinstance(container, ExternalResourcesManager):
+                    file = container
+                else:
+                    parent = container.parent
+                    if parent is not None:
+                        while parent is not None:
+                            if isinstance(parent, ExternalResourcesManager):
+                                file = parent
+                                break
+                            else:
+                                parent = parent.parent
+                    else:
+                        msg = 'Could not find file. Add container to the file.'
+                        raise ValueError(msg)
```

Comment:
This code appears to be copied three times in this file. @mavaylon1 can you extract this into a separate function, perhaps called `_get_file_from_container(self, container)`